src/modules/components/asset/assetTransfer/assetTransferAddress/assetTransferAddress.tsx

```diff
@@ -0,0 +1,79 @@
+import { type IRequiredCompositeAddress } from '../assetTransfer';
+
+export type TxRole = 'sender' | 'recipient';
+
+export interface IAssetTransferAddressProps {
+    /**
+     * Role of the transaction participant.
+     */
+    txRole: TxRole;
+    /**
+     * Address (& optional ENS Name) of the transaction participant.
+     */
+    participant: IRequiredCompositeAddress;
+    /**
+     * URL of the block explorer.
+     */
+    blockExplorerUrl?: string;
+}
+
+export const AssetTransferAddress: React.FC<IAssetTransferAddressProps> = (props) => {
+    const { participant, blockExplorerUrl, txRole } = props;
+
+    const assembledHref = blockExplorerUrl != null ? `${blockExplorerUrl}/address/${participant.address}` : undefined;
+    const resolvedUserHandle =
+        participant.name != null && participant.name !== ''
+            ? participant.name
+            : addressUtils.truncateAddress(participant.address);
+
+    return (
+        <a
+            href={assembledHref}
+            target="_blank"
+            rel="noopener noreferrer"
+            className={classNames(
+                'flex h-20 items-center space-x-4 border-neutral-100 px-4', //base
+                'hover:border-neutral-200 hover:shadow-neutral-md', //hover
+                'focus:outline-none focus-visible:ring focus-visible:ring-primary focus-visible:ring-offset', //focus
+                'active:border-neutral-300', //active
+                'md:w-1/2 md:p-6', //responsive
+                {
+                    'rounded-t-xl  md:rounded-l-xl md:rounded-r-none': txRole === 'sender', // sender base
+                    'rounded-b-xl  md:rounded-l-none md:rounded-r-xl md:pl-8': txRole === 'recipient', // recipient base
+                },
+                {
+                    'border-x border-t md:border-y md:border-l md:border-r-0': txRole === 'sender', // sender borders
+                    'border-x border-b md:border-y md:border-l-0 md:border-r': txRole === 'recipient', // recipient borders
+                },
+                {
+                    'focus-visible:rounded-t-xl md:focus-visible:rounded-l-xl md:focus-visible:rounded-r-none':
+                        txRole === 'sender', // sender focus
+                    'focus-visible:rounded-b-xl md:focus-visible:rounded-l-none md:focus-visible:rounded-r-xl':
+                        txRole === 'recipient', // recipient focus
+                },
+            )}
+        >
+            <MemberAvatar
+                className="shrink-0"
+                responsiveSize={{ md: 'md' }}
+                ensName={participant.name}
+                address={participant.address}
+            />
+            <div className="flex flex-col">
+                <span className="text-xs font-normal leading-tight text-neutral-500 md:text-sm">
+                    {txRole === 'sender' ? 'From' : 'To'}
+                </span>
+                <div className="grid grid-cols-[auto,1fr] items-center gap-x-1">
```

Comment:
We can probably make the text truncate by simply using flex and min-width: 0, it's a bit clearer than using a grid for a simple layout

Reply:
I was fussing with this, couldn't get the min-width: 0 to pick up before going to grid. I will try again.

Reply:
Nice, got it. Need to go div higher to flex-col with the min-w-0.